video/out/opengl/hwdec_cuda.c

```diff
@@ -129,7 +156,7 @@ static int cuda_create(struct gl_hwdec *hw)
 
     p->hwctx = (struct mp_hwdec_ctx) {
         .type = HWDEC_CUDA,
-        .ctx = cuda_ctx,
+        .ctx = display_ctx,
         .av_device_ref = hw_device_ctx,
```

Comment:
Setting .ctx to a different device than .av_device_ref doesn't make sense?
(Also .ctx is probably unused at least by the new code paths.)